internal/services/logic/logic_app_workflow_resource.go

```diff
@@ -869,7 +869,7 @@ func expandLogicAppWorkflowIPAddressRanges(input []interface{}) *[]workflows.IPA
 }
 
 func expandLogicAppWorkflowOpenAuthenticationPolicy(input []interface{}) *map[string]workflows.OpenAuthenticationAccessPolicy {
-	if len(input) == 0 {
+	if len(input) == 0 || input[0] == nil {
```

Comment:
This check isn't needed here as we're going over the range of the slice just a few lines down. Instead we want to check that [item](https://github.com/hashicorp/terraform-provider-azurerm/pull/22486/files#diff-68b6dc20b40a5082f207007f1b104d2ced8fa43fa0ec3b1c2430f4bf02c468e2R878) isn't nil before we cast to it